include/envoy/buffer/buffer.h

```diff
@@ -477,6 +477,24 @@ class Instance {
 
 using InstancePtr = std::unique_ptr<Instance>;
 
+class Chunker {
+public:
+  virtual ~Chunker() = default;
+
+  /**
+   * Copy data into the buffer.
+   * @param data supplies the data..
+   */
+  virtual void add(absl::string_view data) PURE;
+
+  /**
+   * Report Error, cannot call this after add()
+   * @param code supplies the http code.
+   * @param error_text supplies the error string.
+   */
+  virtual void reportError(Http::Code code, absl::string_view error_text) PURE;
+};
```

Comment:
This looks like the right API, but I was thinking this could be a concrete class inside `admin` for now, rather than a pure API.

Also, the implementation of this is not in this PR, right? I couldn't find it.

Reply:
Got it, I will move it into `include/envoy/server/admin.h`, and implement it in `source/server/admin/admin.cc`.

Reply:
Looks like this is still here.